netmiko/base_connection.py

```diff
@@ -147,6 +154,7 @@ def _read_channel_expect(self, pattern='', re_flags=0):
             if re.search(pattern, output, flags=re_flags):
                 if debug:
                     print("Pattern found: {} {}".format(pattern, output))
+                self.write_to_session_log(output)
```

Comment:
This line can be removed if line#149 referenced self.read_channel(). I'm not sure if there was a good reason for that though so I didn't touch it.